samples/inject-vm-data.yaml

```diff
@@ -0,0 +1,29 @@
+# ###
+# This example shows how to use the orka-full Task to create a single macOS VM
+# with an assigned system serial number and custom vm-metadata.
+# The script provided in the params will be executed inside the VM.
+#
+# You will need to have a secret containing the Orka credentials as well as a
+# secret containing the VM SSH credentials as described in the README.
+# ###
+---
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: inject-vm-data
+spec:
+  taskRef:
+    name: orka-full
+  params:
+    - name: base-image
+      value: catalina-ssh-30G.img
+    - name: vm-metadata
+      value: "items: [{ key: testKey, value: testValue }]"
```

Comment:
Is it possible to provide this as YAML rather than as a string? For example:

```yaml
- name: vm-metadata
  value:
    - key: firstKey
      value: firstValue
    - key: secondKey
      value: secondValue
```

Reply:
I would also have VM metadata as its own example ... that way you can demonstrate in the script how to pull metadata from within the VM using the link-local address with something like `curl http://169.254.169.254/metadata/`. https://orkadocs.macstadium.com/docs/retrieving-vm-metadata

Reply:
yeah yaml should be possible